syncer/syncer.go

```diff
@@ -1740,10 +1738,23 @@ type eventContext struct {
 
 // TODO: Further split into smaller functions and group common arguments into a context struct.
 func (s *Syncer) handleRotateEvent(ev *replication.RotateEvent, ec eventContext) error {
-	if ec.header.Timestamp == 0 || ec.header.LogPos == 0 { // fake rotate event
-		if string(ev.NextLogName) <= ec.lastLocation.Position.Name {
+	failpoint.Inject("MakeFakeRotateEvent", func(val failpoint.Value) {
+		ec.header.LogPos = 0
```

Comment:
set Timestamp?

Reply:
when logPos is 0, this event will be treated as a fake roate event